### ISSUE TITLE
[Feature] Require assets (e.g. images) with webpack

### ISSUE THREAD

**Issue by andys8** [halfzebra/create-elm-app#363]
## TLDR

The project could use a webpack plugin to load assets instead of  - or as an alternative to - [passing them from JavaScript via flags](https://github.com/halfzebra/create-elm-app/blob/master/template/README.md#adding-images-and-fonts)

## Explanation

@jhnns helped me create a webpack plugin to "require" an asset path in an Elm project and leverage webpack loaders. We came up with this [proof of concept](https://github.com/andys8/poc-elm-asset-webpack-loader). It uses an ejected `create-elm-app` and adds the webpack loader.

The loader itself is available as [`elm-asset-webpack-loader`](https://github.com/andys8/elm-asset-webpack-loader).

## Usage

```elm
img [ src "require:src/assets/logo.svg" ] []
```

### Result
![image](https://user-images.githubusercontent.com/13085980/62086109-ad420b00-b25d-11e9-9922-44341377ff67.png)

## Syntax

The `require:` syntax can be discussed. I liked it, but I'm open to other ideas.
It could also be configured, but with `create-elm-app` most people will likely stick to the default.

## Implementation

The loader is searching for strings in Elm files already compiled to JavaScript. It replaces the pattern and relies on string replacement. But it'll be more robust than to rely on Elm internals and to support different compiler modes (debug, optimize) and versions (0.18, 0.19, 0.19.1).

## Alternatives and prior art

* [`elm-assets-loader`](https://github.com/NoRedInk/elm-assets-loader) is a comparable approach, and is more sophisticated. The package is marked as deprecated.
* [`babel-elm-assets-plugin`](https://github.com/cultureamp/babel-elm-assets-plugin) is using babel. It mentions to disable Elm optimization. This could be because it's using a `String` `type alias`.

I'm happy to discuss changes to `elm-asset-webpack-loader` or go with an alternative solution or integrate it directly into `create-elm-app` to achieve the goal to use webpack loaders and to likely fail at compile time.

#106

**Comment by jhnns** [comment]
I think I like the prefix `webpack://` better 😁. It's very explicit and mimics the protocol prefix. Happy bikeshedding! 🎉

**Comment by andys8** [comment]
@halbzebra What's your opinion on this feature? Do you want me to come up with a pull request?

**Comment by halfzebra** [comment]
Hello Andy!

First of all, sorry it took me so long to get back to you. This feature makes a lot of sense and I like it a lot. Passing imates using flags is a silly workaround and I have never liked it.

I think it would be a nice addition to Create Elm App, you are very welcome to work on a PR! :+1:

How would the error look if Elm code contained a reference to a missing image?

**Comment by andys8** [comment]
Nice, I'll try to come up with a PR soon.

An error will look like this:

![image](https://user-images.githubusercontent.com/13085980/66721448-03a5fb00-ee04-11e9-858c-7c2c1256c184.png)

![image](https://user-images.githubusercontent.com/13085980/66721456-0dc7f980-ee04-11e9-86b1-c026db9de0a3.png)
